internal/services/audittrail/event_data_source.go

```diff
@@ -0,0 +1,354 @@
+package audittrail
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	audittrailSDK "github.com/scaleway/scaleway-sdk-go/api/audit_trail/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/locality/regional"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/types"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/verify"
+)
+
+func DataSourceEvent() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: DataSourceEventsRead,
+		Schema: map[string]*schema.Schema{
+			"organization_id": {
+				Type:             schema.TypeString,
+				Description:      "ID of the organization containing the Audit Trail events.",
+				Optional:         true,
+				ValidateDiagFunc: verify.IsUUID(),
+			},
+			"region": regional.Schema(),
+			"project_id": {
+				Type:             schema.TypeString,
+				Description:      "ID of the project containing the Audit Trail events.",
+				Optional:         true,
+				ValidateDiagFunc: verify.IsUUID(),
+			},
+			"resource_type": {
+				Type:        schema.TypeString,
+				Description: "Type of the scaleway resources associated with the listed events",
+				Optional:    true,
+				ValidateDiagFunc: func(i any, p cty.Path) diag.Diagnostics {
+					resourceTypeValues := audittrailSDK.ResourceType("").Values()
+
+					resourceTypeStringValues := make([]string, 0, len(resourceTypeValues))
+					for _, resourceTypeValue := range resourceTypeValues {
+						resourceTypeStringValues = append(resourceTypeStringValues, resourceTypeValue.String())
+					}
+
+					return verify.ValidateStringInSliceWithWarning(resourceTypeStringValues, "resourceType")(i, p)
+				},
+			},
+			"resource_id": {
+				Type:             schema.TypeString,
+				Description:      "ID of the Scaleway resource associated with the listed events",
+				Optional:         true,
+				ValidateDiagFunc: verify.IsUUID(),
+			},
+			"product_name": {
+				Type:        schema.TypeString,
+				Description: "Scaleway product associated with the listed events in a hyphenated format",
+				Optional:    true,
+			},
+			"service_name": {
+				Type:        schema.TypeString,
+				Description: "Name of the service of the API call performed",
```

Comment:
Is there a list of those service_name? It could be helpful to provide hinting

Reply:
There appears to be no public list, and they're only pattern validated (no enum on sdk side). 
I have contacted the team to see if they could add a list to their docs for those entries. 
In the meantime I built a list from the sources and I included it in the .mds.